src/core/db/manager.py

```diff
@@ -790,21 +889,72 @@ def has_base_privilege(login, repo_base, privilege):
         return result
 
     @staticmethod
-    def has_repo_privilege(login, repo_base, repo, privilege):
+    def has_repo_db_privilege(login, repo_base, repo, privilege):
+        """
+        Returns a bool describing whether the bool user has the DATABASE
+        privilege passed in the argument. (i.e. Usage)
+
+        Relies on database role management, so this is a pretty straightforward
+        call
+        """
+        repo = repo.lower()
+        repo_base = repo_base.lower()
         with _superuser_connection(repo_base) as conn:
-            result = conn.has_repo_privilege(
+            result = conn.has_repo_db_privilege(
                 login=login, repo=repo, privilege=privilege)
         return result
 
     @staticmethod
+    def has_repo_file_privilege(login, repo_base, repo, privilege):
+        """
+        Returns a bool describing whether or not a user has the FILE privilege
+        passed in the argument. (i.e. 'read')
+
+        """
+        repo = repo.lower()
+        repo_base = repo_base.lower()
+
+        # users always have privileges for their own files
+        if login == repo_base:
+            return True
+
+        # get the collaborator objets for that repo and repo base
+        collaborators = Collaborator.objects.filter(
```

Comment:
updated in 3e91d744b84c5f623e4ff5a65a3d82ebdd7cd0bf. Cool filter + comprehension